resources/metadata/detectExecuteScan.yaml

```diff
@@ -519,6 +519,26 @@ spec:
           - type: vaultSecret
             name: golangPrivateModulesGitTokenVaultSecret
             default: golang
+      - name: scanImages
+        description: If images found in the cpe, they will also be scanned
+        type: "bool"
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        default: false
+      - name: containerDistro
+        description: Distro of the container that is scanned
+        type: "string"
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        default: "ubuntu"
```

Comment:
could remove the default value ?, so as to not start a side car in non docker scans and users should add this param when scanImages is set to true we can add addtional condition in the param 
```

 mandatoryIf:
          - name: scanImages
            value: true
```

Reply:
@anilkeshav27 Let me see if I understand it correctly. Like it is now, there would always be a sidecar container started and you think this is significant overhead, right?

With the change you propose, it would be a mandatory parameter if and only if `scanImages` is set to true. Does the `mandatoryIf` already exist? How is the UX when `scanImages` is set to true but no sidecar selected, is there an obvious message for users what they should do?

Reply:
`mandatoryIf ` as i know will only effect the documentation, but UX can be incorporated in the step logs and message if one cannot find a value for `containerDistro ` 

the UX may become more confusing if adding a default value causes the wrong side car to start  for Distro other than ubuntu . this may have consequences during the scan

Reply:
> the UX may become more confusing if adding a default value causes the wrong side car to start for Distro other than ubuntu .

But your proposed UX (if I get it right) would be even worse imho.
- the user would add `scanImages` and start the pipeline
- the pipeline would fail after a long time because no `containerDistro` is found

Reply:
> * the pipeline would fail after a long time because no `containerDistro` is found

i would prefer this behaviour rather than having a wrong scan run because the default `ubuntu` side car started and instead a `alpine` was needed . its better to have no scan results and a clear failure message than an incorrect scan and confusing  / or even potentially no scan results .

Reply:
I've got an alternative idea. What about we remove the `scanImages` boolean parameter and bind the decision to scan images in the step to the presence of one of the sidecars?

Reply:
Without default of course. The `containerDistro` parameter would just be used for both, the opt-in decision and setting up the step correctly for the distro being used.

Reply:
We combined `scanImages` and `containerDistro` to one parameter `scanContainerDistro` with no default. So by default, no images are scanned and no sidecar is added. If you provide a distro, the right sidecar is added and the images from `cpe` are scanned. @anilkeshav27 That should improve the ux a bit, right?

Reply:
thank you, that looks really good now !!!!